runtime/lua/vim/lsp/inline_completion.lua

```diff
@@ -0,0 +1,243 @@
+---@param context? lsp.InlineCompletionContext
+local function request(bufnr, context)
+  context = context or { triggerKind = 2 }
+
+  vim.lsp.buf_request_all(bufnr, ms.textDocument_inlineCompletion, function(client)
+    ---@type lsp.InlineCompletionParams
+    return {
+      context = context,
+      textDocument = util.make_text_document_params(bufnr),
+      position = util.make_position_params(nil, client.offset_encoding).position,
+    }
+  end, handler)
+end
+
+---@class vim.lsp.inline_completion.JumpOpts
+---@inlinedoc
+---
+---(default: current)
+---@field bufnr? integer
+---
+---(default: 1)
+---@field count? integer
+---
+---(default: true)
+---@field wrap? boolean
+
+---@param opts vim.lsp.inline_completion.JumpOpts
+function M.jump(opts)
+  opts = opts or {}
+  local bufnr = vim._resolve_bufnr(opts.bufnr)
+  local bufstate = bufstates[bufnr]
+  if not bufstate then
+    return
+  end
+
+  local count = opts.count or 1
+  local wrap = opts.wrap or true
+
+  local items = bufstate.items
+  local active = bufstate.current + count
+  active = wrap and (active - 1) % #items + 1 or math.min(math.max(1, active), #items)
+  api.nvim_buf_del_extmark(bufnr, namespace, bufstate.extmark)
+  bufstate.extmark = set_extmark(bufnr, items[active], (' (%d/%d)'):format(active, #items))
+  bufstate.current = active
+end
+
+---@param bufnr? integer
+function M.trigger(bufnr)
+  bufnr = vim._resolve_bufnr(bufnr)
+  local bufstate = bufstates[bufnr]
+  if not bufstate then -- Trigger
+    ---@type lsp.InlineCompletionContext
+    local context = { triggerKind = 1 }
+    request(bufnr, context)
+  else -- Accept
+    local item = bufstate.items[bufstate.current]
+    local text = item.insertText
+    local range = item.range
+    if type(text) ~= 'string' then
+      vim.snippet.expand(text.value)
+    elseif range then
+      util.apply_text_edits({ { newText = text, range = range } }, bufnr, 'utf-16')
+    end
+    clear(bufnr)
+  end
+end
+
+---@generic T: function
+---@param f T
+---@param timeout integer
+---@return T
+local function debunce(f, timeout)
+  ---@type uv.uv_timer_t?
+  local timer = nil
+  return function(...)
+    local args = { ... }
+    if timer then
+      vim.uv.timer_stop(timer)
+      timer:close()
+      timer = nil
+    end
+    timer = assert(vim.uv.new_timer())
+    vim.uv.timer_start(
+      timer,
+      timeout,
+      0,
+      vim.schedule_wrap(function()
+        if timer then
+          vim.uv.timer_stop(timer)
+          timer:close()
+          timer = nil
+        end
+        f(unpack(args))
+      end)
+    )
+  end
+end
+
+---@param bufnr integer
+local function automatic_request(bufnr)
+  request(bufnr, { triggerKind = 2 })
+end
+
+---(default: true)
+---@param enable? boolean
+---
+---(default: current)
+---@param bufnr? integer
+function M.enable(enable, bufnr)
+  enable = enable == nil or enable
+  bufnr = vim._resolve_bufnr(bufnr)
+  local debounced_request = debunce(automatic_request, 100)
```

Comment:
We already have `next_debounce` infrastructure in `_changetracking.lua`. Can we reuse this part of the code?
 
Ref: `client.flags.debounce_text_changes`